RevenueCatUI/Templates/V2/Components/Button/BottomSheetView.swift

```diff
@@ -0,0 +1,238 @@
+    ///   - screenHeightPercentage: The height of the sheet as a percentage of the screen height.
+    ///     Defaults to 0.33333 (one-third of the screen height).
+    ///   - tapOutsideToDismiss: A Boolean value that determines whether tapping outside the sheet
+    ///     dismisses it. Defaults to `true`.
+    init(
+        backgroundColor: Color = Self.systemBackgroundColor,
+        screenHeightPercentage: CGFloat = 0.33333,
+        tapOutsideToDismiss: Bool = true
+    ) {
+        self.backgroundColor = backgroundColor
+        self.screenHeightPercentage = screenHeightPercentage
+        self.tapOutsideToDismiss = tapOutsideToDismiss
+    }
+
+    private static var systemBackgroundColor: Color {
+        #if os(watchOS)
+        Color.black
+        #else
+        return Color(UIColor.systemBackground)
+        #endif
+    }
+}
+
+/// A view modifier that presents content in a sheet-like interface.
+///
+/// This modifier handles the presentation and dismissal of a sheet view, including
+/// the animation and tap-to-dismiss behavior.
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct BottomSheetOverlayModifier<SheetContent: View>: ViewModifier {
+    /// A binding to a Boolean value that determines whether the sheet is presented.
+    let isPresented: Binding<Bool>
+
+    /// The configuration for the sheet.
+    let config: BottomSheetConfig
+
+    /// A closure that creates the content of the sheet.
+    let sheetContent: () -> SheetContent
+
+    @State private var sheetHeight: CGFloat = 0
+
+    func body(content: Content) -> some View {
+        modifierBody(content: content)
+            .ignoresSafeArea()    }
+
+    @ViewBuilder
+    private func modifierBody(
+        content: Content
+    ) -> some View {
+        ZStack {
+            content
+            // Invisible tap area that covers the screen
+            if isPresented.wrappedValue && config.tapOutsideToDismiss {
+                Color.clear
+                    .contentShape(Rectangle())
+                    .onTapGesture {
+                        isPresented.wrappedValue = false
+                    }
+            }
+
+            // Sheet content
+            VStack {
+                Spacer()
+                if isPresented.wrappedValue {
+                    BottomSheetView(
+                        backgroundColor: config.backgroundColor,
+                        height: self.sheetHeight
+                    ) {
+                        sheetContent()
+                    }
+                    .transition(.move(edge: .bottom))
+                }
+            }
+            .background(
+                GeometryReader { proxy in
+                    Color.clear
+                        .onAppear {
+                            self.sheetHeight = proxy.size.height * config.screenHeightPercentage
+                        }
+                }
+            )
+            .animation(.spring(duration: 0.25), value: isPresented.wrappedValue)
```

Comment:
Nit: Can we make this `0.3` or `0.35`? This seems a little fast at the moment

Reply:
Changed to 0.35!